catroid/src/main/java/org/catrobat/catroid/stage/ScreenshotSaver.kt

```diff
@@ -0,0 +1,102 @@
+/*
+ * Catroid: An on-device visual programming system for Android devices
+ * Copyright (C) 2010-2019 The Catrobat Team
+ * (<http://developer.catrobat.org/credits>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.catrobat.catroid.stage
+
+import android.graphics.Bitmap
+import android.util.Log
+import com.badlogic.gdx.Gdx
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.launch
+import org.catrobat.catroid.common.Constants
+import java.io.File
+import java.io.IOException
+
+class ScreenshotSaver(
+    private val folder: String,
+    private val width: Int,
+    private val height: Int
+) {
+
+    companion object {
+        private val TAG = ScreenshotSaver::class.java.simpleName
+        private const val IMAGE_QUALITY = 100
+    }
+
+    fun saveScreenshotAndNotify(
+        data: ByteArray,
+        fileName: String,
+        screenshotCallback: (success: Boolean) -> Boolean
+    ) {
+        GlobalScope.launch(Dispatchers.IO) {
```

Comment:
The question is if, when and how the saving of a screenshot should be cancelled. The current implementation just implies that it should keep on running while the app is alive (which might be perfectly fine).

Otherwise we could give ScreenshotSaver a local coroutine scope and a cancel() function or we could get rid of the callback altogether and make saveScreenshot suspend.
The closest lifecycle aware class is StageActivity, so this would require some refactoring.

Reply:
You are right, it's fine as it is 👍